Expensify/App
# Discourage creation of new workspaces when they already exist

Thanks for the review!

### Details
We want to discourage users from repeatedly creating multiple Workspaces in error (or intentionally, as part of a non-ideal workflow). To do so, we simply hide the "New workspace" option under the Global Create button/menu after a workspace has been created and only reveal it if the user does not have any Workspaces.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/176927

### Tests
As a user without a free workspace:

1. Click the "+" Global Create button
2. Click "New workspace"
3. Close the Workspace modal that pops up
4. Click the "+" Global Create button and verify that you do not see the "New workspace" option

* Caveat: the Part 2 issue: https://github.com/Expensify/Expensify/issues/176926 is what's responsible for allowing deletion of workspaces and updating the policy cache afterwards. Therefore, at the moment, in order to test the return of the menu option after deleting the workspace through OldDot, you must force-refresh (browser + desktop) or sign in & out in order to see the change in the menu options.

### QA Steps
As a user without a free workspace:

1. Click the "+" Global Create button
2. Click "New workspace"
3. Close the Workspace modal that pops up
4. Click the "+" Global Create button and verify that you do not see the "New workspace" option

### Tested On


### Screenshots

#### Web / Desktop
![2021-09-15_17-29-23 (2)](https://user-images.githubusercontent.com/2438855/133530514-61f0e1a6-ce3f-416d-afc7-108103a10e5e.gif)

#### Mobile Web
![2021-09-15_18-00-27 (1)](https://user-images.githubusercontent.com/2438855/133532636-528b0b38-ed1e-4adc-91c3-7a11f2779824.gif)

#### Desktop
![2021-09-15_17-38-51 (1)](https://user-images.githubusercontent.com/2438855/133531139-6d2d600e-0f43-4f29-ae28-aca5690ea8bd.gif)

#### iOS
![2021-09-15_17-48-38 (1)](https://user-images.githubusercontent.com/2438855/133531738-280a196b-ab9f-48e0-b033-d49d4b716bb5.gif)

#### Android
![2021-09-15_18-18-25 (1)](https://user-images.githubusercontent.com/2438855/133533914-fa1d219a-f18c-4391-aa8d-cebe4612fdcd.gif)
